workers/unity/Packages/com.improbable.gdk.core/GameObjectRepresentation/ReaderWriterFactory.cs

```diff
@@ -1,12 +1,66 @@
-﻿using System;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Improbable.Gdk.Core.MonoBehaviours;
+using Unity.Entities;
+using UnityEngine;
 
 namespace Improbable.Gdk.Core
 {
-    public static class ReaderWriterFactory
+    public class ReaderWriterFactory
     {
-        public static object CreateReaderWriter(uint componentId)
+        private const string ComponentIdAttributeNotFoundError
+            = "ReaderWriterCreator found with no Component ID attribute.";
+        private const string NoReaderWriterCreatorFoundForComponentIdError
+            = "No ReaderWriterCreator found for given ComponentId.";
+
+        private readonly EntityManager entityManager;
+        private readonly ILogDispatcher logger;
+        private readonly Dictionary<uint, IReaderWriterCreator> componentIdToReaderWriterCreator = new Dictionary<uint, IReaderWriterCreator>();
+
+        public ReaderWriterFactory(EntityManager entityManager, ILogDispatcher logger)
+        {
+            this.entityManager = entityManager;
+            this.logger = logger;
+
+            FindReaderWriterCreators();
+        }
+
+        private void FindReaderWriterCreators()
         {
-            throw new NotImplementedException();
+            var readerWriterCreatorTypes = AppDomain.CurrentDomain.GetAssemblies()
+                .SelectMany(assembly => assembly.GetTypes())
+                .Where(type => typeof(IReaderWriterCreator).IsAssignableFrom(type) && type.IsClass)
```

Comment:
Any chance of finding `abstract` classes here?

Reply:
I think we don't expect abstract Creators to exist, but just in case it's a good idea to add `&& !type.IsAbstract`

Reply:
Added `!type.IsAbstract` predicate.